src/compiler/checker.ts

```diff
@@ -9737,8 +9774,12 @@ module ts {
             let staticType = <ObjectType>getTypeOfSymbol(symbol);
             let baseTypeNode = getClassBaseTypeNode(node);
             if (baseTypeNode) {
+                if (!isSupportedHeritageClauseElement(baseTypeNode)) {
+                    error(baseTypeNode.expression, Diagnostics.Only_type_references_are_currently_supported_in_a_class_extends_clauses);

```

Comment:
can we mention that it can be identifiers\qualified names with optional type arguments?